redfish-core/lib/log_services.hpp

```diff
@@ -3346,4 +3407,983 @@ inline void requestRoutesPostCodesEntry(App& app)
+                     "HardwareIsolation/Actions/"
+                     "LogService.ClearLog"}}}};
+}
+
+/**
+ * @brief Workaround to handle DCM (Dual-Chip Module) package for Redfish
+ *
+ * This API will make sure processor modeled as dual chip module, If yes then,
+ * replace the redfish processor id as "dcmN-cpuN" because redfish currently
+ * does not support chip module concept.
+ *
+ * @param[in] dbusObjPath - The D-Bus object path to return the object instance
+ *
+ * @return the object instance with it parent instance id if the given object
+ *         is a processor else the object instance alone.
+ */
+inline std::string
+    getIsolatedHwItemId(const sdbusplus::message::object_path& dbusObjPath)
+{
+    std::string isolatedHwItemId;
+
+    if ((dbusObjPath.filename().find("cpu") != std::string::npos) &&
+        (dbusObjPath.parent_path().filename().find("dcm") != std::string::npos))
+    {
+        isolatedHwItemId = std::string(dbusObjPath.parent_path().filename() +
+                                       "-" + dbusObjPath.filename());
+    }
+    else
+    {
+        isolatedHwItemId = dbusObjPath.filename();
+    }
+    return isolatedHwItemId;
+}
+
+/**
+ * @brief API used to fill the respective redfish resource "Id" in the given
+ *        redfishUri by using the given dbus object path which represents the
+ *        given redfishUri and assign the filled redfish uri into
+ *        "OriginOfCondition" property of LogEntry schema.
+ *
+ * @param[in] asyncResp - The redfish response to return.
+ * @param[in] entryJsonIdx - The json entry index to add isolated hardware
+ *                            details in the appropriate entry json object.
+ * @param[in] dbusObjPath - The DBus object path which represents redfishUri.
+ * @param[in] redfishUri - The redfish uri to fill with its "Id".
+ *
+ * @return The redfish response with "OriginOfCondition" property of
+ *         LogEntry schema if success else nothing in redfish response.
+ *
+ * @note - This function will get the associated redfish uri prefix of "Id"
+ *         ("<str>") and look at the predefined redfish list (which will
+ *         contains the pair <DBus Interfaces (which represents redfish
+ *         resource), RedfishUri> of elements) to get the dbus interfaces and
+ *         that is used to get the dbus objects which will compare with the
+ *         given dbus object path to fill the respective uri Id in the
+ *         given redfishUri.
+ *       - This function will run recursively until filling all Ids in given
+ *         given redfishUri.
+ */
+inline void
+    fillRedfishUriWithIds(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                          const size_t entryJsonIdx,
+                          const sdbusplus::message::object_path& dbusObjPath,
+                          std::string& redfishUri)
+{
+
+    std::string uriIdPattern{"<str>"};
+    if (auto uriIdPos = redfishUri.rfind(uriIdPattern);
```

Comment:
I would prefer not auto here..  like we do other places in bmcweb https://github.com/openbmc/bmcweb/search?q=rfind
std::size_t